### DISCUSSION TITLE
Auto Generate Boilerplate App Code

### DISCUSSION THREAD

**Discussion by avan989** [nasa/cFS#389] [Q&A]
Does cfs comes with an way to auto generate boilerplate code for application? Or is there an tool that already does this?

for example: python autogenapp.py appname
and in the app folder contains boilerplate directory structure and files.

**Comment by skliper** [comment]
I think there are ~4 different methods folks use, 3 of which were presented at a recent lunch and learn.

1. Start from https://github.com/nasa/skeleton_app, global replace names and add what you need
2. Start from https://github.com/nasa/sample_app and https://github.com/nasa/sample_lib, global replace names, etc
3. There's a tool developed by JSC (I can't find the link, maybe @tngo67 can help?)
4. There's a tool developed by GSFC (I can't find that link either, maybe @ejtimmon can help?)

Of the above approaches, sample_app/lib is in CI so it's current (compiles, loads, passes unit test).  I'm not sure if the others are actively maintained or not.
